### DISCUSSION TITLE
Connection is established only after couple tries of connecting to the websocket server.

### DISCUSSION THREAD

**Discussion by breakboyandre** [uNetworking/uWebSockets.js#631] [Q&A]
Hello. I'm facing an issue and I dont know how to find what i'm doing wrong. 
Problem:
Sometimes websocket client can't connect to the websocket server with the first try. Sometimes it needs more than 2 tries to connect. Here's video of my problem:

https://user-images.githubusercontent.com/13194235/138395608-bc508437-7d45-4173-8f27-0b3a41cc6a27.mp4

On the server side, "upgrade" and "open" callbacks are not reaching, so i can't find the reason of errors.

Any ideas what can be wrong?

Browser: Chrome 94.0.4606.81, tried with Firefox 93.0, same thing
uWebsSockets.js: 19.3.0

**Comment by Wyzix33** [comment]
What is your reconnect function? 
maybe you try to reconnect to fast, 
also you should try setting a non SSL and see if it's the same problem `const app = uWS./*SSL*/App({` 
also check your server response time and try to disable any firewall/antivirus you may be running

**Comment by breakboyandre** [comment]
> What is your reconnect function? maybe you try to reconnect to fast, also you should try setting a non SSL and see if it's the same problem `const app = uWS./*SSL*/App({` also check your server response time and try to disable any firewall/antivirus you may be running

I updated uWebSockets.js to v20.0.0 and everything is fine so far. But it feels like the problem is not solved. Fells like connection drops depend on my server workload or network ping. I'll watch this situation and text here if problem be back again. 
Thank you.

**Comment by animaonline** [comment]
We're also experiencing similar issues, were you able to resolve this?

**Comment by Wyzix33** [comment]
Maybe try to set a timeout to to your browser socket connection, something like this: https://stackoverflow.com/a/51547396/5660086 , you can try and change the timeout to 3000-4000

**Comment by gitcatrat** [comment]
1. Are you using load balancing, gateway or something like that?
2. Are you seeing same issue if you add regular GET/POST route?

**Comment by animaonline** [comment]
I fixed my problem with help from this post: https://stackoverflow.com/questions/32777760/how-to-fix-missing-an-intermediate-chain-certificate-in-nodejs

**Comment by OmGuptaIND** [comment]
Any Updates on this, guys ??
One critical bug: I hope we can get some solution, or we need to start living with it and handle this behaviour. 

One more observation, connection is prone to getting closed even after first time the connection is established 

I guessing its related to something related to reverse proxy etc. but getting more clarity will make it super helpful to decide what to do next

**Comment by uNetworkingAB** [comment]
Nobody is going to be able to get any kind of information from this kind of high level "my stack of random things is not working". The original video literally gives no information about anything at all. Its all implied.

This is not how you build things. Start by getting your minimal functionality working properly and then add from that. Nobody is going to debug your whole unidentified heap of various components based on "uuh this is not working".

We don't have any such issue with connections in the library so it's not here you report

**Comment by OmGuptaIND** [comment]
You are correct @uNetworkingAB. I was attempting to get if anyone solved there own problem which is somewhat related to mine and that might help me debug my issues faster. 

Again not implying it's with an issue with the library.
